consensus/src/processes/pruning_proof/mod.rs

```diff
@@ -922,18 +944,20 @@ impl PruningProofManager {
                 }
             }
 
+            // Need to ensure this does the same 2M+1 depth that block_at_depth does
             if has_required_block
-                && (root == self.genesis_hash || ghostdag_store.get_blue_score(selected_tip).unwrap() >= required_level_depth)
+                && (root == self.genesis_hash || ghostdag_store.get_blue_score(selected_tip).unwrap() > required_level_depth)
             {
                 break Ok((ghostdag_store, selected_tip, root));
             }
 
             tries += 1;
             if finished_headers {
-                panic!("Failed to find sufficient root for level {level} after {tries} tries. Headers below the current depth of {required_level_0_depth} are already pruned")
+                warn!("Failed to find sufficient root for level {level} after {tries} tries. Headers below the current depth of {required_level_0_depth} are already pruned. Trying anyway.");
```

Comment:
Why not panic?

Reply:
During the first sync with another node that isn't on this version, you'd get something like:

```
2024-07-04 00:31:03.146-06:00 [INFO ] Validating level 24 from the pruning point proof (2002 headers)
2024-07-04 00:31:03.580-06:00 [INFO ] Validating level 23 from the pruning point proof (2002 headers)
2024-07-04 00:31:04.006-06:00 [INFO ] Validating level 22 from the pruning point proof (2002 headers)
2024-07-04 00:31:04.431-06:00 [INFO ] Validating level 21 from the pruning point proof (2002 headers)
2024-07-04 00:31:04.868-06:00 [INFO ] Validating level 20 from the pruning point proof (2054 headers)
2024-07-04 00:31:05.316-06:00 [INFO ] Validating level 19 from the pruning point proof (2101 headers)
2024-07-04 00:31:05.758-06:00 [INFO ] Validating level 18 from the pruning point proof (2108 headers)
2024-07-04 00:31:06.223-06:00 [INFO ] Validating level 17 from the pruning point proof (2002 headers)
2024-07-04 00:31:06.659-06:00 [INFO ] Validating level 16 from the pruning point proof (2002 headers)
2024-07-04 00:31:07.096-06:00 [INFO ] Validating level 15 from the pruning point proof (2001 headers)
2024-07-04 00:31:07.539-06:00 [INFO ] Validating level 14 from the pruning point proof (2020 headers)
2024-07-04 00:31:07.994-06:00 [INFO ] Validating level 13 from the pruning point proof (2001 headers)
2024-07-04 00:31:08.415-06:00 [INFO ] Validating level 12 from the pruning point proof (2001 headers)
2024-07-04 00:31:08.867-06:00 [INFO ] Validating level 11 from the pruning point proof (2001 headers)
2024-07-04 00:31:09.288-06:00 [INFO ] Validating level 10 from the pruning point proof (2001 headers)
2024-07-04 00:31:09.695-06:00 [INFO ] Validating level 9 from the pruning point proof (2001 headers)
2024-07-04 00:31:10.119-06:00 [INFO ] Validating level 8 from the pruning point proof (2001 headers)
2024-07-04 00:31:10.545-06:00 [INFO ] Validating level 7 from the pruning point proof (2001 headers)
2024-07-04 00:31:10.962-06:00 [INFO ] Validating level 6 from the pruning point proof (2001 headers)
2024-07-04 00:31:11.391-06:00 [INFO ] Validating level 5 from the pruning point proof (2001 headers)
2024-07-04 00:31:11.804-06:00 [INFO ] Validating level 4 from the pruning point proof (2001 headers)
2024-07-04 00:31:12.219-06:00 [INFO ] Validating level 3 from the pruning point proof (2001 headers)
2024-07-04 00:31:12.636-06:00 [INFO ] Validating level 2 from the pruning point proof (2001 headers)
2024-07-04 00:31:13.053-06:00 [INFO ] Validating level 1 from the pruning point proof (2001 headers)
2024-07-04 00:31:13.492-06:00 [INFO ] Validating level 0 from the pruning point proof (2001 headers)
```

And along with that you'd see these:

```
2024-07-04 00:31:27.716-06:00 [INFO ] Building the proof which was just applied (sanity test)
2024-07-04 00:31:27.986-06:00 [WARN ] Failed to find sufficient root for level 23 after 1 tries. Retrying again to find with depth 2560000
2024-07-04 00:31:28.231-06:00 [WARN ] Failed to find sufficient root for level 15 after 1 tries. Headers below the current depth of 5000 are already pruned. Trying anyway.
2024-07-04 00:31:28.293-06:00 [WARN ] Failed to find sufficient root for level 13 after 1 tries. Headers below the current depth of 2500 are already pruned. Trying anyway.
2024-07-04 00:31:28.321-06:00 [WARN ] Failed to find sufficient root for level 12 after 1 tries. Headers below the current depth of 2500 are already pruned. Trying anyway.
2024-07-04 00:31:28.350-06:00 [WARN ] Failed to find sufficient root for level 11 after 1 tries. Headers below the current depth of 2500 are already pruned. Trying anyway.
2024-07-04 00:31:28.379-06:00 [WARN ] Failed to find sufficient root for level 10 after 1 tries. Headers below the current depth of 2500 are already pruned. Trying anyway.
2024-07-04 00:31:28.409-06:00 [WARN ] Failed to find sufficient root for level 9 after 1 tries. Headers below the current depth of 2500 are already pruned. Trying anyway.
2024-07-04 00:31:28.441-06:00 [WARN ] Failed to find sufficient root for level 8 after 1 tries. Headers below the current depth of 2500 are already pruned. Trying anyway.
2024-07-04 00:31:28.473-06:00 [WARN ] Failed to find sufficient root for level 7 after 1 tries. Headers below the current depth of 2500 are already pruned. Trying anyway.
2024-07-04 00:31:28.502-06:00 [WARN ] Failed to find sufficient root for level 6 after 1 tries. Headers below the current depth of 2500 are already pruned. Trying anyway.
2024-07-04 00:31:28.533-06:00 [WARN ] Failed to find sufficient root for level 5 after 1 tries. Headers below the current depth of 2500 are already pruned. Trying anyway.
2024-07-04 00:31:28.565-06:00 [WARN ] Failed to find sufficient root for level 4 after 1 tries. Headers below the current depth of 2500 are already pruned. Trying anyway.
2024-07-04 00:31:28.596-06:00 [WARN ] Failed to find sufficient root for level 3 after 1 tries. Headers below the current depth of 2500 are already pruned. Trying anyway.
2024-07-04 00:31:28.628-06:00 [WARN ] Failed to find sufficient root for level 2 after 1 tries. Headers below the current depth of 2500 are already pruned. Trying anyway.
2024-07-04 00:31:28.658-06:00 [WARN ] Failed to find sufficient root for level 1 after 1 tries. Headers below the current depth of 2500 are already pruned. Trying anyway.
2024-07-04 00:31:28.849-06:00 [WARN ] Locally built proof for level 0 does not match the applied one
```

The new way of finding root takes up some safety margin, but the old nodes don't so we wouldn't otherwise be able to sync if we panic.

Reply:
So maybe add a comment that explains it?

Reply:
Revisiting this idea maybe we should panic `finished_headers` is true and `has_required_block` is false. In the above example where we sync with an older node without margins `has_required_block` must still be true anyway.